src/platform/posix/tcp_ip_channel.c

```diff
@@ -469,7 +469,17 @@ static void TcpIpChannel_free(NetworkChannel *untyped_self) {
   self->super.close_connection((NetworkChannel *)self);
 }
 
-NetworkChannelState TcpIpChannel_get_state(NetworkChannel *untyped_self) {
+static uint32_t TcpIpChannel_get_dest_channel_id(NetworkChannel *untyped_self) {
+  TcpIpChannel *self = (TcpIpChannel *)untyped_self;
+
+  if (self->server) {
+    return self->client;
+  } else {
+    return self->fd;
+  }
+}
```

Comment:
This depends on how `get_dest_channel_id` is to be used by the runtime. In fact, I think it might not be used by the runtime at all, only by the lower levels of the NetworkChannel implementation. This suggests that it might not be the right decision to have this function as part of the generic NetworkChannel API, rather it some bookkeeping that must be done in the different concrete implementations?